app/views/foreman_bootdisk/generic_boot_no_dhcp.erb

```diff
@@ -0,0 +1,41 @@
+#!ipxe
+#
+# This template is generic, but it will chainload to Foreman so expects the
+# host to be registered already.
+#
+# It loops through all interfaces asking if this is the one that the host
+# was registered with, then prompting for IP settings to configure
+# the interface with.
+#
+
+# loop over net*
+<% (0..32).each do |i| -%>
+:net<%= i %>
+isset ${net<%= i -%>/mac} || goto no_nic
+echo net<%= i -%> is a ${net<%= i -%>/chip} with MAC ${net<%= i -%>/mac}
+echo -n Enter this interface name to boot from it (net<%= i -%>|n):  && read interface
```

Comment:
I'd perhaps put this outside of the loop, so print all interfaces and then ask the user to type in the one they want.

Reply:
Hi Dominic,

The reason why this was in there, is to verify that this is the right nic to boot from, by comparing the MAC address.
and in this, the right option will show in "()" as an example "Enter this interface name to boot from it (net0|n): "

Would you still prefer to have it outside of the loop?
If that is the case, I think I would need to put some words around it to remember MAC address and netX from outputs.

Reply:
Let me elaborate what (I think) Dominic think. There is a NIC naming convention in iPXE: net0, net1, net2 ... netN. You want the user to enter "netN" to select the proper interface. Now, users will likely identify NIC from MAC addresses. So first use the loop to print out network devices, something like:

```
Found AA:BB:CC:DD:EE:F0 as nic0 
Found AA:BB:CC:DD:EE:F1 as nic1 
Found AA:BB:CC:DD:EE:F2 as nic2
```

And then carry on as usual:

```
Enter interface name to configure: net1
Enter IP address: _
```